subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationQueue.java

```diff
@@ -164,11 +165,11 @@ public void run() {
         }
 
         private void runBuildOperation() {
-            BuildOperationWorkerRegistry.Completion workerLease = owner.operationStart();
+            WorkerLeaseRegistry.WorkerLeaseCompletion workerLease = owner.startChild();
```

Comment:
I think we definitely could (and probably should) but there is at least one place where we start/finish the lease outside of a try/finally block (`ForkingTestClassProcessor`) and I'm not sure I understand why that was done this way.  I've created a follow up issue to tackle this separately.